extensions/amp-story/1.0/test/test-amp-story-quiz.js

```diff
@@ -199,5 +232,24 @@ describes.realWin(
       expect(variables[AnalyticsVariable.STORY_REACTION_RESPONSE]).to.equal(0);
       expect(variables[AnalyticsVariable.STORY_REACTION_TYPE]).to.equal(0);
     });
+
+    it('should update the quiz when the user has already reacted', async () => {
+      populateStandardQuizContent(win, ampStoryQuiz.element);
+      ampStoryQuiz.buildCallback();
+      await ampStoryQuiz.layoutCallback();
+
+      const quizElement = ampStoryQuiz.getQuizElement();
+      const quizOptions = quizElement.querySelectorAll(
+        '.i-amphtml-story-quiz-option'
+      );
+
+      // Mock a successful data retrieval.
+      ampStoryQuiz.handleSuccessfulDataRetrieval_(getMockReactionData());
```

Comment:
There are many rules to writing good unit tests and no one knows all of them, but one that's interesting to master is the "Testing via public APIs".
What it means is that you should only call public methods, as they usually test your class end to end, and aren't likely to change (it'd be a breaking change). Not testing private methods gives you more freedom to refactor your code without having to update all the tests.

What it means for this test is that, instead of manually calling this request handler, you could stub the `requestService` so it returns what you want. This way, you test that the component actually sent the request, and reacted properly, without depending on any implementation detail that could change with any PR.

```
    env.sandbox.stub(requestService, 'executeRequest').resolves(getMockReactionData());
```